src/sentry/models/project.py

```diff
@@ -175,9 +176,12 @@ def save(self, *args, **kwargs):
                     reserved=RESERVED_PROJECT_SLUGS,
                     max_length=50,
                 )
-            super().save(*args, **kwargs)
-        else:
-            super().save(*args, **kwargs)
+
+        snowflake_redis_key = "project_snowflake_key"
+        self.save_with_snowflake_id(
+            snowflake_redis_key, lambda: super(Project, self).save(*args, **kwargs)
+        )
```

Comment:
FYI for other reviews, this `SnowflakeIdMixin` idea was a solution that I pair-programmed with @maxiuyuan and I'm open to feedback.

I still kind of like it, but feel just a bit iffy on injecting the lambda callback like that.